numba/cuda/tests/cudapy/test_powi.py

```diff
@@ -61,6 +71,32 @@ def test_powi(self):
         kernel[1, A.shape](A, power, Aout)
         self.assertTrue(np.allclose(Aout, A ** power))
 
+    def test_powi_int64(self):
+        dec = cuda.jit(void(float64[:, :], int64, float64[:, :]))
+        kernel = dec(cu_mat_power_v2)
+
+        A = np.arange(10, dtype=np.float64).reshape(2, 5)
+        Aout = np.empty_like(A)
+
+        for power in range(1, 11):
+            power = np.int64(power)
+            kernel[1, A.shape](A, power, Aout)
+            result = np.allclose(Aout, A ** power)
+            self.assertTrue(result, f"Failed on int64 power {power}")
```

Comment:
* A combination of `np.allclose` and `self.assertTrue` is odd - the `np.testing` module contains functions such as `assert_allclose()` etc that report the numerical issue detected.
* If you want to report the test data for which a test failed, you can use `self.subTest()` (as in other uses in the test suite) rather than adding your own message.
* Using `allclose()` for a test of integer arithmetic seems surprising - why not check for exact equality?